### ISSUE TITLE
Test suite is incomplete for Django 3.2 compatibility

### ISSUE THREAD

**Issue by pulse-mind** [noripyt/django-cachalot#180]
This issue refer to https://github.com/noripyt/django-cachalot/issues/175. 

Hello,

I tried to use your branch @Natureshadow on my application so then the application started but then I've got some errors during execution. Some parts of the applications are working fine but some not. It was working fine on Django 3.1
I was using this in my requirements: 
```
-e git+https://github.com/Natureshadow/django-cachalot.git@remove-dependency-check#egg=django-cachalot
```
Note that if I use django3.2 + I disable cachalot in INSTALLED_APPS by using comments then it works.
So there is at least bug using 3.2 and django-cachalot.

I am doing a simple aggregate on a queryset:
```
...
purchaseorder_lines= Line.objects.select_related('parent', 'parent__source').filter(parent__status='validated')[:20]
purchaseorder_lines.aggregate(total1=Sum('total1'),total2=Sum('total2'))
...
```

This is the end of the stacktrace: 
```
...
  File "D:\Perso\Projects\xxxxx\xxx\items\admin.py", line 994, in change_view
    purchaseorder_totals = purchaseorder_lines.aggregate(
  File "D:\Perso\VirtualEnv\xxxxx\lib\site-packages\django\db\models\query.py", line 399, in aggregate
    return query.get_aggregation(self.db, kwargs)
  File "D:\Perso\VirtualEnv\xxxxx\lib\site-packages\django\db\models\sql\query.py", line 511, in get_aggregation
    result = compiler.execute_sql(SINGLE)
  File "d:\perso\virtualenv\xxxxx\src\django-cachalot\cachalot\monkey_patch.py", line 29, in inner
    return original(compiler, *args, **kwargs)
  File "d:\perso\virtualenv\xxxxx\src\django-cachalot\cachalot\monkey_patch.py", line 84, in inner
    table_cache_keys = _get_table_cache_keys(compiler)
  File "d:\perso\virtualenv\xxxxx\src\django-cachalot\cachalot\utils.py", line 191, in _get_table_cache_keys
    for t in _get_tables(db_alias, compiler.query)]
  File "d:\perso\virtualenv\xxxxx\src\django-cachalot\cachalot\utils.py", line 172, in _get_tables
    _get_tables_from_sql(connections[db_alias], query.subquery))
  File "d:\perso\virtualenv\xxxxx\src\django-cachalot\cachalot\utils.py", line 99, in _get_tables_from_sql
    return {t for t in connection.introspection.django_table_names()
  File "d:\perso\virtualenv\xxxxx\src\django-cachalot\cachalot\utils.py", line 100, in <setcomp>
    if t in lowercased_sql}
TypeError: argument of type 'bool' is not iterable```

**Comment by Andrew-Chen-Wang** [comment]
This may be due to the CI just not running due to no credits. Will try to fix with #173. In the meantime, closing this as a duplicate.

**Comment by fabiocaccamo** [comment]
@Andrew-Chen-Wang have you any estimation about releasing a new version with django 3.2 compatibility?
Thank you for your great contribution on this library.

**Comment by Andrew-Chen-Wang** [comment]
@fabiocaccamo Apologies for not being able to get this done sooner like the 3.1 update. Swamped with finals and work. I would like to merge #177 as I've reviewed it, but the CI is stuck. Once I get the CI working again (probably today), then we be GTG.

**Comment by Andrew-Chen-Wang** [comment]
So I'm out of finals. Like always, Django's changed something with Subqueries again and it's broken the test cases. I'm going to be occupied for this weekend, so if anyone would like to help get this pushed out quickly, please contribute at #181 via comments and suggestions. The hope is I can release cachalot 2.4.0 with full support for Django 3.2 by next week, but I'm also looking to get subquery to be nailed down without future, spontaneous errors. Anyhow, further discussion of that can be in the PR.

**Comment by Andrew-Chen-Wang** [comment]
@fabiocaccamo @pulse-mind @Natureshadow apologies for being super late with this fix but **v2.4.0 is now out**. There were a lot of changes to the Subquery API that I could not for the life of me figure out until today. Again, super super sorry for being this late!

**Comment by fabiocaccamo** [comment]
@Andrew-Chen-Wang thank you very much!!
